test/integration/short/speculative-execution-simulator-tests.js

```diff
@@ -0,0 +1,448 @@
+'use strict';
+var assert = require('assert');
+var responseErrorCodes = require('../../../lib/types').responseErrorCodes;
+var simulacron = require('../simulacron');
+var utils = require('../../../lib/utils');
+
+var Client = require('../../../lib/client.js');
+var ConstantSpeculativeExecutionPolicy = require('../../../lib/policies/speculative-execution').ConstantSpeculativeExecutionPolicy;
+var OrderedLoadBalancingPolicy = require('../../test-helper').OrderedLoadBalancingPolicy;
+
+describe('Client', function() {
+  describe('With ConstantSpeculativeExecutionPolicy', function () {
+    var delay = 100;
+    var clientOptions = {
+      policies: { 
+        speculativeExecution: new ConstantSpeculativeExecutionPolicy(delay, 3), 
+        loadBalancing: new OrderedLoadBalancingPolicy()
+      }
+    };
+    var setupInfo = simulacron.setup('3', { clientOptions: clientOptions });
+    var client = setupInfo.client;
+    var cluster = setupInfo.cluster;
+    var query = "select * from data";
+    
+    var assertQueryCount = queryCounter(query, cluster);
+
+    it('should not start speculative executions if query is non-idempotent', function (done) {
```

Comment:
This is just a start.  I looked at the java-driver 4.x tests for inspiration for the most part.  There's a lot of code duplication among the tests, will clean that up.